pkg/workflow/engineimpl.go

```diff
@@ -247,21 +251,25 @@ func (e *EngineImpl) InvokeWithInputAndConfig(
 
 			// prepare logger
 			prefix := fmt.Sprintf("%s:%d", id.Host, e.invocationCounter)
-			e.mu.Unlock()
```

Comment:
I think, we'd in general only need a read-lock (and an RWMutex). Nothing blocking though. Should the lock maybe even start earlier, e.g. in L240 to avoid init races?